R/dfm.R

```diff
@@ -52,301 +50,72 @@ dfm.default <- function(x, ...) {
     check_class(class(x), "dfm")
 }
 
-#' @export
-dfm.character <- function(x, ...) {
```

Comment:
These should be marked `.Defunct` rather than just removed. I am considering using the **lifecycle** package for deprecations and removals, see https://r-pkgs.org/lifecycle.html#sec-lifecycle-breaking-change-pros-cons, as this works really well for informing users of exactly what was removed and when. This will save a lot of clueless SO questions because it's more informative.

Reply:
I agree we want to remove them in v4, the issue is simply how. No real need to do this here. Although I know how tempting it is. I promise a major cleanup is coming!

Reply:
See [7fdea44](https://github.com/quanteda/quanteda/pull/2251/commits/7fdea44cf9de1c24afb00e57c9516f26c8d6b909), this Defuncts them in the most minimal way possible. Also adds an argument to `check_class()` so that defunct methods are not listed as valid when the warning is printed for invalid object class inputs.

Reply:
@koheiw if you're happy with this for now, you or I can mark this conversation as resolved and then merge the PR.